### PR TITLE
"exposing host port" example

### LINKED ISSUE #1136: Example needed: exposing host port
It would be nice to add an example of `Testcontainers.exposeHostPorts` method. It is very helpful when you're running something inside a container but want to access a port from the host.

Let's say:
1) start a very simple Spring Boot app in the test with some HTML page
1) use `Testcontainers.exposeHostPorts(serverPort)` to expose it to the containers network.
1) start `BrowserWebDriverContainer`
1) run some very simple Selenium assertion of the HTML page

Here is a test for the feature:
https://github.com/testcontainers/testcontainers-java/blob/bcecd5cd9f9325517fd45db585312df2624315bb/core/src/test/java/org/testcontainers/containers/ExposedHostTest.java

### PR DESCRIPTION
**Pull request by stsefanenko**
Update example, added expose host port. Fixes #1136

### CODE REVIEW

File: examples/selenium-container/src/test/java/SeleniumContainerTest.java [hunk 1]
```diff
@@ -14,20 +21,30 @@
...
+        Testcontainers.exposeHostPorts(8080);
+    }
 
     @Test
     public void simplePlainSeleniumTest() {
         RemoteWebDriver driver = chrome.getWebDriver();
 
-        driver.get("https://wikipedia.org");
-        List<WebElement> searchInput = driver.findElementsByName("search");
+        driver.get("http://host.testcontainers.internal:" + 8080 + "/foo");
```
Comment (bsideup) [inline examples/selenium-container/src/test/java/SeleniumContainerTest.java]:
```suggestion
        driver.get("http://host.testcontainers.internal:8080/foo");
```

File: examples/selenium-container/src/test/java/SeleniumContainerTest.java [hunk 2]
```diff
@@ -14,20 +21,30 @@
...
 public class SeleniumContainerTest {
 
+    @LocalServerPort
+    private int port;
+
     @Rule
     public BrowserWebDriverContainer chrome = new BrowserWebDriverContainer()
-                                                    .withCapabilities(new ChromeOptions())
-                                                    .withRecordingMode(RECORD_ALL, new File("target"));
+        .withCapabilities(new ChromeOptions())
```
Comment (bsideup) [inline examples/selenium-container/src/test/java/SeleniumContainerTest.java]: could you please revert this change?

File: examples/selenium-container/src/test/java/SeleniumContainerTest.java [hunk 3]
```diff
@@ -14,20 +21,30 @@
 /**
  * Simple example of plain Selenium usage.
  */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = DemoApplication.class, webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
```
Comment (bsideup) [inline examples/selenium-container/src/test/java/SeleniumContainerTest.java]: Please change from `DEFINED_PORT` to `RANDOM_PORT` and use `LocalServerPort`'s value when calling `exposeHostPorts`

File: examples/selenium-container/src/main/resources/templates/foo.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE HTML>
```
Comment (bsideup) [inline examples/selenium-container/src/main/resources/templates/foo.html]: I think you can move it to `src/main/resources/static/foo.html` and simply query it with `/foo.html`, so that you don't need to create a controller

File: examples/selenium-container/src/test/java/TestFooController.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = DemoApplication.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class TestFooController {
```
Comment (bsideup) [inline examples/selenium-container/src/test/java/TestFooController.java]: I don't think it is needed for this test

File: examples/selenium-container/src/test/java/SeleniumContainerTest.java (hunk 2 above)
Comment (stsefanenko) [reply examples/selenium-container/src/test/java/SeleniumContainerTest.java]: ok

File: examples/selenium-container/src/test/java/TestFooController.java (hunk 5 above)
Comment (stsefanenko) [reply examples/selenium-container/src/test/java/TestFooController.java]: ok, I'll remove it

**Comment by bsideup** [comment]
@stsefanenko great job!
I've left a few comments but it looks good in general and a great addition to our examples, well done 👍

File: examples/selenium-container/src/test/java/SeleniumContainerTest.java (hunk 3 above)
Comment (stsefanenko) [reply examples/selenium-container/src/test/java/SeleniumContainerTest.java]: Updated

File: examples/selenium-container/src/main/resources/templates/foo.html (hunk 4 above)
Comment (stsefanenko) [reply examples/selenium-container/src/main/resources/templates/foo.html]: ok, changed

File: examples/selenium-container/pom.xml [hunk 6]
```diff
@@ -11,40 +11,61 @@
...
-            <scope>provided</scope>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
         <dependency>
-            <groupId>ch.qos.logback</groupId>
-            <artifactId>logback-classic</artifactId>
-            <version>1.2.3</version>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-thymeleaf</artifactId>
```
[the remark this answers is not in the document]
Comment (stsefanenko) [reply examples/selenium-container/pom.xml]: yes. removed it

**Comment by bsideup** [comment]
@stsefanenko merged! 🎉 Thank you for your contribution 👍
